src/mixins/_theme.scss

```diff
@@ -0,0 +1,51 @@
+@use "../design-tokens/colors.yml";
+@use "../design-tokens/opacity.yml";
+@use "../design-tokens/sizes.yml";
+
+/// Defines the default theme, used to determine selectors to output as well
+/// as default arguments.
+/// @type String
+$default-theme: light;
+
+/// @param {String} $theme [$default-theme]
+/// @output Content within the appropriate root-level theme selector
+/// @content
+@mixin props($theme: $default-theme) {
+  @if $theme == $default-theme {
+    :root {
+      @content;
+    }
+  }
+
+  .t-#{$theme} {
+    @content;
+  }
+}
```

Comment:
Nit: Would we output less compiled CSS if we combined these property sets similar to how you did in the mixin below?

```scss
  @if $theme == $default-theme {
    :root,
    .t-#{$theme} {
      @content;
    }
  } @else {
    .t-#{$theme} {
      @content;
    }
  }
```

Reply:
@Paul-Hebert When I did that, I got a `selector-root-no-composition` error from Stylelint. I assumed there was a reason for that and reverted to what I have now.

Because of your question, I attempted to look up _why_ that rule exists. I found that it's actually been removed from the most recent versions of Stylelint.

I found [this discussion](https://github.com/stylelint/stylelint/issues/2226) where it's suggested that it _may_ have existed to promote compatibility with PostCSS plugins for automating custom property fallbacks, which (at least at the time) only worked for `:root` selectors without more complex selectors.

So I guess I can probably combine these selectors again and disable the linting error there?

Reply:
@Paul-Hebert Selectors are now combined and this rule has been disabled in the project's lint config